node/chainSimulator/process/processor.go

```diff
@@ -62,16 +62,51 @@ func (creator *blocksCreator) createHeaderBasedOnRound(round uint64, nonce uint6
 func (creator *blocksCreator) createBlock(header data.HeaderHandler) (data.HeaderHandler, data.BodyHandler, error) {
 	processComponents := creator.nodeHandler.GetProcessComponents()
 	if header.IsHeaderV3() {
-		return processComponents.BlockProcessor().CreateBlockProposal(header, func() bool {
+		newHeader, newBody, err := processComponents.BlockProcessor().CreateBlockProposal(header, func() bool {
 			return true
 		})
+		if err != nil {
+			return nil, nil, err
+		}
+
+		return newHeader, newBody, creator.updateHeader(header)
 	}
 
 	return processComponents.BlockProcessor().CreateBlock(header, func() bool {
 		return true
 	})
 }
 
+func (creator *blocksCreator) updateHeader(header data.HeaderHandler) error {
+	rootHash, err := creator.nodeHandler.GetStateComponents().AccountsAdapter().RootHash()
+	if err != nil {
+		return err
+	}
+
+	switch h := header.(type) {
+	case *dataBlock.HeaderV3:
+		if check.IfNil(h) || h.LastExecutionResult == nil {
+			return nil
+		}
```

Comment:
- maybe check header is nil first thing in the function
- and maybe return some error because header should have execution results for header v3
```suggestion
		if h.LastExecutionResult == nil || len(h.ExecutionResults) == 0 {
			return SOME_ERROR
		}
```

Reply:
refactored